src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,44 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type == Bulldozer.class) {
+            BulldozerProducer producer = new BulldozerProducer();
+            return new ArrayList<>(producer.get());
+        } else if (type == Truck.class) {
+            TruckProducer producer = new TruckProducer();
+            return new ArrayList<>(producer.get());
+        } else if (type == Excavator.class) {
+            ExcavatorProducer producer = new ExcavatorProducer();
+            return new ArrayList<>(producer.get());
+        } else {
+            return null;
```

Comment:
return null is a bad practice. Let's return an empty list instead

Reply:
resolved!